cmd/fscrypt/keys.go

```diff
@@ -179,23 +180,30 @@ func makeKeyFunc(supportRetry, shouldConfirm bool, prefix string) actions.KeyFun
 			if prefix != "" {
 				return nil, ErrNotPassphrase
 			}
-			prompt := fmt.Sprintf("Enter key file for protector %q: ", info.Name())
-			// Raw keys use a file containing the key data.
-			file, err := promptForKeyFile(prompt)
-			if err != nil {
-				return nil, err
-			}
-			defer file.Close()
 
-			fileInfo, err := file.Stat()
-			if err != nil {
-				return nil, err
-			}
+			if keyFileFlag.Value != "" || term.IsTerminal(stdinFd) {
```

Comment:
This function is getting too long.  How about moving this section into a new function, like `func makeRawKey(info actions.ProtectorInfo) (*crypto.Key, error)`?

Reply:
@ebiggers please see below. I hope that's what you meant.